bubblewrap.c

```diff
@@ -518,7 +518,7 @@ switch_to_user_with_privs (void)
   if (prctl (PR_SET_KEEPCAPS, 1, 0, 0, 0) < 0)
     die_with_error ("prctl(PR_SET_KEEPCAPS) failed");
 
-  if (setuid (real_uid) < 0)
+  if (setuid (opt_sandbox_uid) < 0)
```

Comment:
if `--uid` was not specified (which is allowed only with `--unshare-user`), then `opt_sandbox_uid` is set to `real_uid` in `main`.